### ISSUE TITLE
Reconsider deprecation warnings in general, from the end-user point of view

### ISSUE THREAD

**Issue by rubyFeedback** [biopython/biopython#2730]
I am using biopython-1.76, python is 3.8.1, linux version, everything compiled from source.

Since the recent biopython, whenever I start the interactive python interpreter, I get
this:

[GCC 9.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
ty/home/Programs/Python/3.8.1/lib/python3.8/site-packages/Bio/Alphabet/__init__.py:22: 
PendingDeprecationWarning: We intend to remove or replace Bio.Alphabet in 2020, ideally
avoid using it explicitly in your code. Please get in touch if you will be adversely
affected by this.

https://github.com/biopython/biopython/issues/2046
  warnings.warn("We intend to remove or replace Bio.Alphabet in 2020,

I tried to comment to https://github.com/biopython/biopython/issues/2046 but a clever person limited this to collaborators only, so I had to create a new issue here. ;)

So what is the problem?

The thing is ... the above warning is quite annoying. Every start up I see it. There may be ways
to silence it, fair enough, but I just don't think it should be the default really. 

I understand your motivation e. g. to let people know before the upgrade, but I am
mostly just an end user really and have not written that much python altogether
(I am evidently more of a ruby person but I also use python).

My suggestion would be more in general, to not show deprecation warnings
all the time, but instead find some way to show this only once, or perhaps
not show it at all. Perhaps when people install biopython there could
be additional hooks; and/or alternatively you could add some list with
incompatibilities or upcoming changes people could check. These are
just some ideas. I just contest that the idea that on every start up of
the python interactive interpreter the above deprecation should be
shown, is ... a strange assumption to push forward to. It may be that
I also use some biopython code, have not checked yet, but to me the
deprecation warning is much more annoying than knowing that this
will be changed "some time in the future".

Thank you for reading this.

**Comment by peterjc** [comment]
Whatever we do here can't please everyone.

I am unclear about the use case - are you using some script or tool which in turn used Biopython?

**Comment by mdehoon** [comment]
I would agree with @rubyFeedback ; generally DeprecationWarnings don't seem very useful. If we want to alert the user to a change in Biopython with an easy alternative (e.g. if a function moved from module Bio.A to Bio.B), then I would prefer an Exception informing the user how their script should be adjusted. DeprecatingWarnings may be useful only in cases where some code is about to be removed from Biopython, and there is no recommended alternative available. Warnings about future changes can be left out if there is nothing the user should actually do about it. See also PEP 565 (https://www.python.org/dev/peps/pep-0565/) and the Python documentation (https://docs.python.org/3/whatsnew/2.7.html#changes-to-the-handling-of-deprecation-warnings) with some discussion within Python about the pros and cons of deprecation warnings.

**Comment by peterjc** [comment]
In the context of https://docs.python.org/3/whatsnew/2.7.html#changes-to-the-handling-of-deprecation-warnings is it still true that most Biopython users are developers writing their own code? There will be a number of people using it indirectly through other libraries/applications (which is think is how the bug report would describe themselves), and they certainly don't want to see the deprecation warnings.

We introduced ``BiopythonDeprecationWarning`` rather than use the standard library ``DeprecationWarning`` precisely because that was silent by default.

It would be a simple change to make ``BiopythonDeprecationWarning`` a subclass of ``DeprecationWarning``, which I assume would then be silent by default too. If not, a global replace of all our usage would be straightforward (leaving the exception definition in place but labelled as deprecated - people will potentially be using it to silence the warning).

**Comment by MarkusPiotrowski** [comment]
At least on Windows, I can not reproduce this. Did I understand something wrong?
```
...\Python38>python
Python 3.8.1 (tags/v3.8.1:1b293b6, Dec 18 2019, 23:11:46) [MSC v.1916 64 bit (AM
D64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import Bio
>>> Bio.__version__
'1.76'
>>> from Bio import Seq
>>> from Bio import Alphabet
>>> alpha = Alphabet.Alphabet()
>>> alpha
Alphabet()
```

Also, I find it strange that you get a `BiopythonDeprecationWarning` just by starting the Python interpreter?

**Comment by peterjc** [comment]
Good point, for ``Bio.Alphabet`` we are deliberately using ``PendingDeprecationWarning`` which is silent by default .

Something on @rubyFeedback's setup must be showing all warnings (even the silent by default ones).

**Comment by MarkusPiotrowski** [comment]
Cross referencing #2488

**Comment by Gerardwx** [comment]
Is it possible to set PYTHONWARNINGS to ignore the BiopythonDeprecationWarning? We get it when running a third party application,

**Comment by peterjc** [comment]
@Gerardwx try adding ``ignore::BiopythonDeprecationWarning`` or ``ignore::Bio.BiopythonDeprecationWarning`` to your comma-separated list in ``$PYTHONWARNINGS``, see https://docs.python.org/3/library/warnings.html#describing-warning-filters

But please do log an issue with the third party application team, they will need to change something in their code because if the ``BiopythonDeprecationWarning`` is being triggered, their code is going to break in a future Biopython release.

**Comment by Gerardwx** [comment]
export PYTHONWARNINGS=ignore::BiopythonDeprecationWarning

works, thanks.
